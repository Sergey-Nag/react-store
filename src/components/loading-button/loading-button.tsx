import { ButtonHTMLAttributes, ReactNode } from 'react';
import cn from 'classnames';

interface LoadingButtonProps
  extends ButtonHTMLAttributes<HTMLButtonElement> {
  loading: boolean;
  children: ReactNode;
}

export function LoadingButton({ className, loading, children, ...rest }: LoadingButtonProps) {
  const buttonClasses = cn('btn btn-primary', className);

  return (
    <button
      className={buttonClasses}
      {...rest}
    >
      { loading
          ? (<><span className="spinner-border spinner-border-sm me-2" role="status" aria-hidden="true"></span>Loading</>) 
          : children
      }
    </button>
  )
}