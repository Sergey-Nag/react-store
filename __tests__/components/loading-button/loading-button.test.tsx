import { render } from '@testing-library/react';
import { LoadingButton } from '../../../src/components/loading-button/loading-button';

describe('Loading button', () => {
  it('Should render button without loader if the loading property is false', () => {
    const { asFragment } = render(<LoadingButton loading={false}>test</LoadingButton>);

    expect(asFragment()).toMatchSnapshot('without loader');
  });

  it('Should render button with loader if the loading property is true', () => {
    const { asFragment } = render(<LoadingButton loading={true}>test</LoadingButton>);

    expect(asFragment()).toMatchSnapshot('with loader');
  });
});