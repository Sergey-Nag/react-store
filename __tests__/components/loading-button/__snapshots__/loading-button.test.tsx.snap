// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading button Should render button with loader if the loading property is true: with loader 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
  >
    <span
      aria-hidden="true"
      class="spinner-border spinner-border-sm me-2"
      role="status"
    />
    Loading
  </button>
</DocumentFragment>
`;

exports[`Loading button Should render button without loader if the loading property is false: without loader 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
  >
    test
  </button>
</DocumentFragment>
`;
